src/hooks/useMutate.js

```diff
@@ -52,11 +54,22 @@ export default function useMutate({
     const status = response?.status;
     setStatusCode(status);
     if (status === 200 || status === 204) {
-      const queryKeysFromArgs = deriveQueryKeys(mutationArgs);
-      const invalidationKeys = [...queryKeys, ...queryKeysFromArgs];
-      invalidationKeys.forEach(queryKey => {
+      const invalidations = [
+        ...invalidateKeys,
+        ...deriveInvalidateKeys(mutationArgs),
+      ];
+      invalidations.forEach(queryKey => {
         queryClient.invalidateQueries(queryKey);
       });
+
+      const fetches = [
+        ...fetchKeys,
+        ...deriveFetchKeys(mutationArgs),
+      ];
+      fetches.forEach(async queryKey => {
+        await queryClient.refetchQueries(queryKey);
+      });
```

Comment:
@Emily-Ke I'm not sure that's true. I tested by adding a handful of queryKeys to the same mutation, adding the following logs, and running the mutation:

```
      console.log('start foreach');
      fetches.forEach(async queryKey => {
        console.log('fetching ', queryKey);
        await queryClient.refetchQueries(queryKey);
      });
      console.log('end foreach');
```

result:
```
useMutate.js:69 start foreach
useMutate.js:71 fetching  users
useMutate.js:71 fetching  me
useMutate.js:71 fetching  collaborations
useMutate.js:74 end foreach
```

Reply:
I did consider Promise.all but was a little concerned with hitting the server with a ton of simultaneous requests. It _should_ be fine but my confidence is low. Maybe if it feels important we can do some stress-testing after mvp

Reply:
```
console.log('start foreach');
fetches.forEach(async queryKey => {
  console.log('fetching ', queryKey);
  await queryClient.refetchQueries(queryKey);
  console.log('finished fetching ', queryKey);
});
console.log('end foreach');
```
result:
```
useMutate.js:69 start foreach
useMutate.js:71 fetching  (2) ['user', 'guid']
useMutate.js:71 fetching  me
useMutate.js:71 fetching  users
useMutate.js:75 end foreach
useMutate.js:73 finished fetching  (2) ['user', 'guid']
useMutate.js:73 finished fetching  users
useMutate.js:73 finished fetching  me
```
These requests fire almost simultaneously. At the very least the `async/await` keywords are unnecessary since there is no code within the forEach callback function that needs to execute after the refetch has resolved.

If you want the refetches to happen sequentially, apparently a `for...of` loop works:
```
console.log('start foreach');
for (const queryKey of fetches) {
  console.log('fetching ', queryKey);
  await queryClient.refetchQueries(queryKey);
  console.log('finished fetching ', queryKey);
}
console.log('end foreach');
```
result:
```
useMutate.js:69 start foreach
useMutate.js:71 fetching  (2) ['user', 'guid']
useMutate.js:73 finished fetching  (2) ['user', 'guid']
useMutate.js:71 fetching  me
useMutate.js:73 finished fetching  me
useMutate.js:71 fetching  users
useMutate.js:73 finished fetching  users
useMutate.js:75 end foreach
```

Reply:
okay you're totally right. sorry for not seeing it earlier. good news is i think this resolves the loading issue i was seeing earlier too!